Feature

TinWin99 opened the pull request:
Pull request from Tin

Zeraiz-Shabbir committed:
Created initial_setup.xml and budgeting.xml

TinWin99 committed:
I created 3 xml files, limit_exceeded_warning_popup.xml, savings_and_spending_progress.xml, and savings_depletion_warning_popup.xml

KrishnaDOS committed:
- Added BudgetContract.java, BudgetDBHelper.java, BudgetDataSource.java
- Basic database functionality has been implemented successfully. Said functionality includes:
    - Create a database (first time only)
    - Create a table (given a specific table name)
    - Add/remove a statement from the given table

Zeraiz-Shabbir committed:
Tentatively finished add_income_expense.xml

Zeraiz-Shabbir committed:
- Added click events for all activities stemming from InitialSetupActivity.
- Submit button for InitialSetupActivity is only enabled upon entering a starting balance.
- InitialSetupActivity will only appear the first time you launch the app. Upon subsequent launches, it will not appear.
- Added icons to all buttons in activity_initial_setup.xml.
- Created activity_main.xml and implemented RecyclerView which automatically updates upon the start of a new month.

KrishnaDOS commented:
This request has been closed since we found out the right way to fetch from feature branch.

KrishnaDOS commented:
I thought I was closing the pull request but it closed the branch? I'm not sure how this works... anyways I reopened the pull request.

TinWin99 committed:
I created 3 xml files, limit_exceeded_warning_popup.xml, savings_and_spending_progress.xml, and savings_depletion_warning_popup.xml

TinWin99 committed:
Merge branch 'feature' into tin

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/example/budgetplanner/MainActivity.java
#	app/src/main/res/layout/activity_main.xml
#	app/src/main/res/values/strings.xml

KrishnaDOS committed:
- Added SavingsContract.java - To store current savings amount, current balance, and spending limit
- Modified BudgetDBHelper.java and BudgetDataSource.java to accommodate new table contract
- Refactored several method names

KrishnaDOS committed:
- Added editing functionality to the database; user can edit/remove a statement after providing updated values and unique row identifier / serial number
- Added DBTest.java - Contains code to test out database functionality
- Modified BudgetDBHelper.java and BudgetDataSource.java to accommodate editing functionality

KrishnaDOS committed:
- Added Statement.java which stores information about a given statement in the database table
- Modified BudgetDataSource.java to accommodate Statement objects rather than primitive type fields
- Method BudgetDataSource.getStatements created in order to retrieve all statements for the month in question from the respective table in the database

KrishnaDOS committed:
- Added database functionality to feature branch

KrishnaDOS committed:
- Added .xml files to feature branch

KrishnaDOS committed:
- Added a new field to monthly statement information table
    - Stores the frequency of the income or purchase for a given month
- Added a new field to the budgeting table for each month
    - Stores the amount already spent (accumulation of all purchases for a month)
- Updated relevant classes, methods, and objects to accommodate these changes

TinWin99 committed:
Merge remote-tracking branch 'origin/tin' into tin

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/res/values/strings.xml

TinWin99 committed:
Merge branch 'feature' into tin

KrishnaDOS committed:
- Added a new field to monthly statement information table
    - Stores the frequency of the income or purchase for a given month
- Added a new field to the budgeting table for each month
    - Stores the amount already spent (accumulation of all purchases for a month)
- Updated relevant classes, methods, and objects to accommodate these changes

TinWin99 committed:
I created 3 xml files, limit_exceeded_warning_popup.xml, savings_and_spending_progress.xml, and savings_depletion_warning_popup.xml
halfway done implementing dialog for pop up

TinWin99 committed:
Merge remote-tracking branch 'origin/feature' into feature

# Conflicts:
#	app/src/main/java/com/example/budgetplanner/AddIncomeExpenseActivity.java
#	app/src/main/java/com/example/budgetplanner/BudgetingActivity.java

TinWin99 committed:
Remove conflict
savings_and_spending_progress.xml, and savings_depletion_warning_popup.xml
halfway done implementing dialog for pop up
